crates/worker/src/partition_processor_manager.rs

```diff
@@ -995,10 +1001,15 @@ impl PartitionProcessorManager {
                     snapshot_repository,
                 };
 
+                let jitter_start = if sender.is_some() {
+                    Duration::ZERO
+                } else {
+                    Duration::from_millis(rand::rng().random_range(0..10_000))
```

Comment:
With an average added duration of 5s, we probably should also update the `latest_snapshot_check_interval` which is currently set to 5s. With this change, every second check will probably become meaningless because we are still waiting for the jitter.

Reply:
I know that Ahmed prefers to start ad a small prime number like 3ms to avoid resonance frequencies to occur. Here it probably does not matter that much since we limit the number of concurrent operations to 4 any way.

Reply:
I think the other reason why it's probably less impactful here is that we jitter the (0..4) snapshot tasks individually. But it's a good idea anyway, I'll add some small prime number of millis to the peiord. I reduced the check itself to 1s in a follow-up PR, I'll update both.

Reply:
Ended up lifting Ahmed's approach for log trim check of adding 10% jitter to the interval.